_studio/mfx_lib/scheduler/src/mfx_scheduler_core_ischeduler.cpp

```diff
@@ -149,8 +130,74 @@ mfxStatus mfxSchedulerCore::Initialize2(const MFX_SCHEDULER_PARAM2 *pParam)
     }
 
     return MFX_ERR_NONE;
+}
+
+mfxStatus mfxSchedulerCore::SetThreadNum(const void *pComponent, mfxU32 threadNum)
+{
+    MFX_CHECK(pComponent, MFX_ERR_INVALID_HANDLE);
+
+    // If user requested certain number of threads we follow his choice
+    // doing nothing in this function.
+    if (m_param.params.NumThread)
+        return MFX_ERR_NONE;
+
+    std::lock_guard<std::mutex> map_lock(m_pThreadsMapMutex);
+
+    if (threadNum)
+        m_pThreadsMap[pComponent] = threadNum;
+    else
+        m_pThreadsMap.erase(pComponent);
+
+    threadNum = 0; // reuse it now to count number of total threads
+    for (auto it: m_pThreadsMap) {
+        threadNum += it.second;
+    }
+
+    mfxU32 maxParallelism = std::thread::hardware_concurrency();
+    if (threadNum > maxParallelism) threadNum = maxParallelism;
```

Comment:
As stated in https://en.cppreference.com/w/cpp/thread/thread/hardware_concurrency
"Number of concurrent threads supported. If the value is not well defined or not computable, returns"

So maybe add a check for zero in following way:
```
mfxU32 maxParallelism = std::thread::hardware_concurrency();
if (maxParallelism && threadNum > maxParallelism)
     threadNum = maxParallelism;
```